vtex-sites/base.store
# chore: lazy load icons

## What's the purpose of this pull request?
This PR removes all phosphor icons from the app bundle. This improves performance and decreases the size of the final bundle. Doesn't remove the icons from CartSideBar because it is already lazy-loaded.

Motivation: Phosphor react icons are really good, but they are heavy and they were on the final app bundle. So lazy loading them will remove them from the final bundle.
As you can see, in red is the phosphor react components
**Before**:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/15680320/155357065-602f0112-7602-4ba6-94b9-0b95684e24b1.png">

**After**:
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/15680320/155356939-e8024c83-4d2a-44ee-9710-f77e802b3263.png">

Also, the final document decreased the size from 21.7kb to 20.0kb according to the https://pagespeed.web.dev
Finally, it reduced the app size 
**from**:
<img width="316" alt="image" src="https://user-images.githubusercontent.com/15680320/155357845-cd160e49-f271-44c5-bf4f-ee2ba6232240.png">
**to**:
<img width="315" alt="image" src="https://user-images.githubusercontent.com/15680320/155357915-96a6a9d3-4a62-4e5d-88e8-138be1dbed00.png">


⚠️ This PR changes the **lighthouse** settings.

## How does it work?
Part 1:
Created a LazyLoad component that renders the Phosphor icon lazily. The webpack will create new chunks for each lazy icon.
Part 2:
Change the lighthouse assertions, to allow fetch more scripts. This is needed because part 1 creates more scripts for each icon.

## How to test it?
Check the evidence:
https://preview-358--base.preview.vtex.app/bundle-analyser.html
https://base.vtex.app/bundle-analyser.html

Also, compare the preview with the main URL to ensure that's the icons are rendering at the same size and positions.

## References
https://reactjs.org/docs/code-splitting.html

## Checklist
TBD

- [] CHANGELOG entry added
